voxblox/include/voxblox/utils/planning_utils.h

```diff
@@ -1,119 +1,161 @@
-void clearSphereAroundPoint(Layer<EsdfVoxel>* layer, const Point& center,
-                            const FloatingPoint radius);
-
-/* TODO(mereweth@jpl.nasa.gov) - implement something that makes more sense for
- * projective distance. This allows marking a known area as free or clear and
- * preserving that marking across TSDF updates
- */
-
-/* This function sets all voxels within a Euclidean distance of the center to
- * have the specified distance values
- */
-inline void fillSphereAroundPoint(Layer<EsdfVoxel>* layer,
-                                  const Point& center,
-                                  const FloatingPoint radius) {
-  CHECK_NOTNULL(layer);
-  // search a cube with side length 2*radius
-  const FloatingPoint voxel_size = layer->voxel_size();
-  for (FloatingPoint x = -radius; x <= radius; x += voxel_size) {
-   for (FloatingPoint y = -radius; y <= radius; y += voxel_size) {
-     for (FloatingPoint z = -radius; z <= radius; z += voxel_size) {
-       Point point(x, y, z);
-
-       // check if point is inside the spheres radius
-       const FloatingPoint radius_squared_norm = radius * radius;
-       if (point.squaredNorm() <= radius_squared_norm) {
-         // convert to global coordinate
-         point += center;
-
-         const Block<EsdfVoxel>::Ptr block_ptr =
-             layer->allocateBlockPtrByCoordinates(point);
-         const VoxelIndex voxel_index =
-             block_ptr->computeTruncatedVoxelIndexFromCoordinates(point);
-         EsdfVoxel& voxel = block_ptr->getVoxelByVoxelIndex(voxel_index);
-         const Point voxel_center =
-             block_ptr->computeCoordinatesFromVoxelIndex(voxel_index);
-
-         const Point voxel_center_vec = voxel_center - point;
-
-         // how far is voxel from center of filled sphere
-         const FloatingPoint new_distance = sqrt(voxel_center_vec.squaredNorm());
-
-         if ((!voxel.observed)               ||
-             (new_distance < voxel.distance)) {
-           voxel.distance = new_distance;
-           voxel.observed = true;
-           voxel.fixed = true;
-           block_ptr->updated() = true;
-           block_ptr->has_data() = true;
-         }
-       }
-     }
-   }
-  }
-}
-
-/* This function sets all voxels within a Euclidean distance of the center to
- * have the specified distance values
- */
-inline void clearSphereAroundPoint(Layer<EsdfVoxel>* layer,
-                                   const Point& center,
-                                   const FloatingPoint radius) {
-  CHECK_NOTNULL(layer);
-  // search a cube with side length 2*radius
-  const FloatingPoint voxel_size = layer->voxel_size();
-  for (FloatingPoint x = -radius; x <= radius; x += voxel_size) {
-    for (FloatingPoint y = -radius; y <= radius; y += voxel_size) {
-      for (FloatingPoint z = -radius; z <= radius; z += voxel_size) {
+// Gets the indices of all points within the sphere.
+template <typename VoxelType>
```

Comment:
does it make sense to break utils into planning_utils.h and planning_utils_inl.h if you are needing to foward declare some functions?

Reply:
Yep.